src/dumpling-service/dumpling.web/Controllers/DumplingServiceController.cs

```diff
@@ -0,0 +1,190 @@
+        /// At the moment a user will need to upload a single chunk, and then after that they can proceed to upload additional chunks to 
+        /// </summary>
+        /// <param name="owner"></param>
+        /// <param name="targetos"></param>
+        /// <param name="index"></param>
+        /// <param name="filesize"></param>
+        /// <returns></returns>
+        [Route("dumpling/store/chunk/{owner}/{targetos}/{index}/{filesize}")]
+        [HttpPost]
+        public async Task<string> PostDumpChunk(string owner, string targetos, int index, ulong filesize, string displayName = "")
+        {
+            if(index > 0 || filesize > int.MaxValue)
+            {
+                throw new NotSupportedException("We do not support chunked files yet, and the file must be <= 2GB or more specifically, int.MaxValue");
+            }
+
+            if (!Request.Content.IsMimeMultipartContent("form-data"))
+            {
+                throw new HttpResponseException(HttpStatusCode.UnsupportedMediaType);
+            }
+
+
+            await DumplingEventHub.FireEvent(new WebAPIStartUploadChunkEvent());
+
+            /* Handle Upload */
+            // Get or create the blob container
+            var container = DumpStorageController.GetContainerForOwner(owner);
+            await container.CreateIfNotExistsAsync();
+
+            // Create a AzureBlobStorageMultipartProvider and process the request
+            var path = Path.GetTempPath();
+            AzureBlobStorageMultipartProvider streamProvider = new AzureBlobStorageMultipartProvider(container, path);
+
+            await Request.Content.ReadAsMultipartAsync<AzureBlobStorageMultipartProvider>(streamProvider);
+
+            await DumplingEventHub.FireEvent(new WebAPIFinishedUploadChunkEvent());
+
+            /* Meta data handling */
+            var dump_uri = streamProvider.AzureBlobs.First().Location;
+
+            if(displayName == string.Empty)
+            {
+                displayName = Path.GetFileNameWithoutExtension(Path.GetRandomFileName());
+            }
+
+            int dumplingid = await TriageDb.AddDumpAsync(new Dump()
+            {
+                DumpTime = DateTime.Now,
+                DumpPath = dump_uri,
+                DisplayName = displayName,
+                Origin = owner
+            });
+
+            StateTableIdentifier id = new StateTableIdentifier()
+            {
+                Owner = owner,
+                DumplingId = dumplingid.ToString(),
+            };
+
+            // let the dumpling services know about our dumpling
+            await StateTableController.AddOrUpdateStateEntry(new StateTableEntity(id)
+            {
+                DumpRelics_uri = dump_uri,
+                OriginatingOS = targetos
+            });
+
+            await StateTableController.SetState(id, "enqueued");
+
+            await AnalysisTopicController.EnqueueAnalysisWork(owner, dumplingid.ToString(), targetos, dump_uri);
```

Comment:
I think you want to switch the order of this.  If EnqueuAnalysisWork fails than this leaves the dump in an enqueued state when it hasn't been enqueued.